Issue 4306 - Bean validation should fail if @Valid field is not introspected

wetted opened the pull request:
Issue 4306 - Bean validation should fail if @Valid field is not introspected
- emit validation failure when `@Valid` alone is applied to a container of elements where the element type is not `@Introspected`.
- this does not address cases with additional constraint(s), since that might still be legal if the element type is supported by those constraints directly (e.g. `@Valid @Size(min=1, max=2) List<String> strs`), even though something like `@Valid @Size(min=1, max=2) List<NotIntrospected> oops` might be bogus.

wetted committed:
Issue 4306 - Bean validation should fail if @Valid field is not introspected
- emit validation failure when @Valid alone is applied to a container of elements where the element type is not @Introspected.
- this does not address cases with additional constraint(s), since that might still be legal if the element type is supported by those constraints directly (e.g. @Valid @Size(min=1, max=2) List<String> strs), even though something like @Valid @Size(min=1, max=2) List<NotIntrospected> oops might be bogus.

jameskleeh commented on validation/src/main/java/io/micronaut/validation/validator/DefaultValidator.java:
```diff
@@ -1331,6 +1320,14 @@ public void keyedValue(String nodeName, Object key, Object keyedValue) {
 
         final BeanIntrospection<Object> beanIntrospection = getBeanIntrospection(propertyValue);
+        AnnotationMetadata annotationMetadata = cascadeProperty.getAnnotationMetadata();
+        if (beanIntrospection == null && annotationMetadata.getDeclaredAnnotationNames().equals(
```
`@Valid` is not annotated with `@Constraint` so you can just look for the presence of that annotation instead of comparing the annotation names

wetted committed:
Issue 4306 - Bean validation should fail if @Valid field is not introspected
- emit validation failure when @Valid alone is applied to a container of elements where the element type is not @Introspected.
- this does not address cases with additional constraint(s), since that might still be legal if the element type is supported by those constraints directly (e.g. @Valid @Size(min=1, max=2) List<String> strs), even though something like @Valid @Size(min=1, max=2) List<NotIntrospected> oops might be bogus.

wetted committed:
Issue 4306 - Bean validation should fail if @Valid field is not introspected
- emit validation failure when @Valid alone is applied to a container of elements where the element type is not @Introspected.
- this does not address cases with additional constraint(s), since that might still be legal if the element type is supported by those constraints directly (e.g. @Valid @Size(min=1, max=2) List<String> strs), even though something like @Valid @Size(min=1, max=2) List<NotIntrospected> oops might be bogus.
